### PR TITLE
[ktx] Fix CMake configure error

### LINKED ISSUE #18556: [ktx] CMake configure error
**Describe the bug**

**Environment**
- OS: Windows 10 Build 21390
- Compiler: Visual C++ 2019

**To Reproduce**
1. ./vcpkg install ktx:x64-windows
2. In a Visual Studio CMake project,
```
find_package(Ktx CONFIG REQUIRED)
target_link_libraries(main PRIVATE KTX::ktx)
```
3. Project ->Generate cache

**Expected behavior**
CMake configuration should succeed.

**Failure logs** 
CMake outputs:
```
Severity	Code	Description	Project	File	Line	Suppression State
Error		CMake Error at xxx/vcpkg/installed/x64-windows/share/ktx/KtxTargets.cmake:80 (message):
  The imported target "KTX::ktx" references the file

     "xxx/vcpkg/installed/x64-windows/debug/bin/ktx.dll"

  but this file does not exist.  Possible reasons include:

  * The file was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and contained

     "xxx/vcpkg/installed/x64-windows/share/ktx/KtxTargets.cmake"

  but not all the files it references.		xxx/vcpkg/installed/x64-windows/share/ktx/KtxTargets.cmake	80	

```

### PR DESCRIPTION
**Pull request by JonLiu1993**
**Describe the pull request**

- #### What does your PR fix?  
  Fixes #18556

````
Severity	Code	Description	Project	File	Line	Suppression State
Error		CMake Error at xxx/vcpkg/installed/x64-windows/share/ktx/KtxTargets.cmake:80 (message):
  The imported target "KTX::ktx" references the file

     "xxx/vcpkg/installed/x64-windows/debug/bin/ktx.dll"

  but this file does not exist.  Possible reasons include:

  * The file was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and contained

     "xxx/vcpkg/installed/x64-windows/share/ktx/KtxTargets.cmake"

  but not all the files it references.		xxx/vcpkg/installed/x64-windows/share/ktx/KtxTargets.cmake	80	
````

I have test the usage locally:
````
PS F:\test9\test\build> F:\test9\vcpkg\downloads\tools\cmake-3.21.1-windows\cmake-3.21.1-windows-i386\bin\cmake.exe .. -G "Visual Studio 16 2019" -DCMAKE_TOOLCHAIN_FILE=F:/test9/vcpkg/scripts/buildsystems/vcpkg.cmake -DVCPKG_TARGET_TRIPLET=x64-windows
-- Selecting Windows SDK version 10.0.19041.0 to target Windows 10.0.19042.
-- The C compiler identification is MSVC 19.29.30137.0
-- The CXX compiler identification is MSVC 19.29.30137.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.29.30133/bin/Hostx64/x64/cl.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.29.30133/bin/Hostx64/x64/cl.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring done
-- Generating done
-- Build files have been written to: F:/test9/test/build

PS F:\test9\test\build> F:\test9\vcpkg\downloads\tools\cmake-3.21.1-windows\cmake-3.21.1-windows-i386\bin\cmake.exe --build .
Microsoft (R) Build Engine version 16.11.2+f32259642 for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

  Checking Build System
  Building Custom Rule F:/test9/test/CMakeLists.txt
  main.cpp
  test.vcxproj -> F:\test9\test\build\Debug\test.exe
  Building Custom Rule F:/test9/test/CMakeLists.txt

````

### CODE REVIEW

**Comment by JackBoosY** [comment]
Closing this PR since it seems that no progress is being made. Please reopen if work is still being done.

File: ports/ktx/fix-links-target.patch [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+         find_package(zstd CONFIG REQUIRED)
+-        target_link_libraries(
+-            ${lib}
+-            $<IF:$<TARGET_EXISTS:zstd::libzstd_shared>,zstd::libzstd_shared,zstd::libzstd_static>
+-        )        
++        if (BUILD_SHARED_LIBS)       
++            target_link_libraries(${lib} PRIVATE zstd::libzstd_shared)
++        else()
++            target_link_libraries(${lib} PRIVATE zstd::libzstd_static)
++        endif()
```
Comment (dg0yt) [inline ports/ktx/fix-links-target.patch:17]: This change is wrong. `BUILD_SHARED_LIBS` is controlling output for port ktx here, but not the zstd input. The original approach looks valid, and I don't see a hint that it failed.

Comment (JonLiu1993) [reply ports/ktx/fix-links-target.patch:17]: @dg0yt ,  I see this issue has been fixed, I will close this pr, thanks for the heads up

**Comment by JonLiu1993** [comment]
Fixed in https://github.com/microsoft/vcpkg/pull/23942.

**Pull request closed without merging**